### PR TITLE
Adapt block creation for Altair

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Adapt the block creation process to allow for setting the `SyncAggregate` field in Altair blocks.  Currently the sync committee logic isn't implemented so it's not being set but wanting to get a sense of whether this approach to supporting the setting of different fields works.

The key question is whether we should go this way and keep `BeaconBlockBody` as immutable or should follow the approach of `BeaconState` and introduce a `MutableBeaconBlockBody` and that mechanism for building blocks.  The key challenge being that while the `spec` module contains the block data structures and `BlockFactory`, it doesn't have access to the `OperationPool` and other sources of information that go into a block, so code outside the spec needs to adjust based on whether the block needs a `SyncAggregate` or not.

## Fixed Issue(s)
#3648 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/BeaconBlockBodyContent.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import tech.pegasys.teku.spec.datastructures.blocks.Eth1Data;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.altair.SyncAggregate;
+import tech.pegasys.teku.spec.datastructures.operations.Attestation;
+import tech.pegasys.teku.spec.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.spec.datastructures.operations.Deposit;
+import tech.pegasys.teku.spec.datastructures.operations.ProposerSlashing;
+import tech.pegasys.teku.spec.datastructures.operations.SignedVoluntaryExit;
+import tech.pegasys.teku.ssz.SszList;
+
+public interface BeaconBlockBodyContent {
```
Comment (Nashatyrev) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/BeaconBlockBodyContent.java]: May be `BeaconBlockBodyBuilder` would be more suitable? 
That's definitely the builder pattern and the `BeaconBlockBodyContent` slightly confused me at the first glance.

**Review by Nashatyrev — APPROVED** [review]
The key feature of `MutableBeaconState` is that it inherits `BeaconState` and one may use existing data to derive new values and then access updated values (state transitions). But this feature yields unsafety as well: leaking non-threadsafe `MutableBeaconState` as thread-safe `BeaconState` is dangerous (that's why disallowing `BeaconState.createMutableCopy()`)

I guess this is not the case for the `BeaconBlock`, i.e. you would likely not need existing block data to modify the block. If the latter is correct then I would stick with your current `BeaconBlockBodyContent` (I would call it `Builder` though) solution as more explicit and safe

Everything else looks good 👍

**Comment by Nashatyrev** [comment]
In other words using potential `MutableBeaconBlock` as a _builder_ is kind of overkill and potentially not safe. 
`MutableBeaconState` was an unavoidable trade off cause a builder pattern just didn't suite there

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/BeaconBlockBodyContent.java (hunk 1 above)
Comment (ajsutton) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/BeaconBlockBodyContent.java]: Yeah I went back and forth between strictly following the builder pattern and calling it `Builder` or not so strictly following it.  Wound up following the builder pattern but didn't go back and rename. :)

**Comment by ajsutton** [comment]
> In other words using potential `MutableBeaconBlock` as a _builder_ is kind of overkill and potentially not safe.
> `MutableBeaconState` was an unavoidable trade off cause a builder pattern just didn't suite there

Good, that's justification enough for me to not follow that pattern. Apart from some tidy up around names etc, I was fairly happy with where this landed, just worried about introducing a new pattern if the existing one for states made sense to use.  But sounds like they're different patterns for clearly different use cases so all good.

**Pull request merged**